nat-lab/tests/utils/network_switcher/network_switcher_windows.py

```diff
@@ -1,16 +1,68 @@
 import config
+import re
 from .network_switcher import NetworkSwitcher
+from dataclasses import dataclass
+from typing import List, Optional
 from utils.connection import Connection
 from utils.process import ProcessExecError
 
-VAGRANT_DEFAULT_IF = "Ethernet"  # 10.0.2.0/24
-PRIMARY_IF = "Ethernet 2"  # 10.55.0.11
-SECONDARY_IF = "Ethernet 3"  # 10.66.0.11
+
+@dataclass
+class Interface:
+    def __init__(self, name: str, ipv4: str) -> None:
+        self.name = name
+        self.ipv4 = ipv4
+
+
+class ConfiguredInterfaces:
+    def __init__(self, default: str, primary: str, secondary: str) -> None:
+        self.default = default
+        self.primary = primary
+        self.secondary = secondary
+
+    @staticmethod
+    async def find(connection: Connection) -> "ConfiguredInterfaces":
+        interfaces = await ConfiguredInterfaces.get_network_interfaces(connection)
+
+        def find_interface(prefix: str) -> str:
+            for interface in interfaces:
+                if interface.ipv4.startswith(prefix):
+                    return interface.name
+            assert False, f"interface not found with prefix `{prefix}`, {interfaces}"
+
+        return ConfiguredInterfaces(
+            find_interface(config.LIBVIRT_MANAGEMENT_NETWORK_PREFIX),
+            find_interface(config.PRIMARY_VM_NETWORK_PREFIX),
+            find_interface(config.SECONDARY_VM_NETWORK_PREFIX),
+        )
```

Comment:
It looks just like constructor

Reply:
Hmm, in my imagination we could pass result of `get_network_interfaces` to it and make it not `async` and thus make it just a constructor, but maybe passing `ConfiguredInterfaces` is somehow nicer than `List[Interface]`? I'm not sure which one would be better TBH

Reply:
Are you talking about something like this?
```
class ConfiguredInterfaces:
    def __init__(self, interface: List[Interface]) -> None:
        def find_interface(prefix: str) -> str:
            for interface in interfaces:
                if interface.ipv4.startswith(prefix):
                    return interface.name
            assert False, f"interface not found with prefix `{prefix}`, {interfaces}"

        self.default = find_interface(config.LIBVIRT_MANAGEMENT_NETWORK_PREFIX)
        self.primary = find_interface(config.PRIMARY_VM_NETWORK_PREFIX)
        self.secondary = find_interface(config.SECONDARY_VM_NETWORK_PREFIX)
```

I find this to be harder to read than the static method. I think at this point its a matter of personal preference.